pkg/build.go

```diff
@@ -150,6 +150,14 @@ func NewImageDependencies(env *BuilderContext, image string, runtime string, bui
 	for _, buildtime := range buildtimes {
 		bt := NormalizeImageTag(env.Expand(buildtime))
 
+		// If the image is prefixed with "library/", remove it for comparisons.
+		// "library/" will be added again during image reference generation.
+		// This prevents duplicate dependencies when reading "library/golang" and
+		// "golang" from the Dockerfile.
+		if strings.HasPrefix(bt, "library/") {
```

Comment:
Does it also apply for non docker-hub image? eg,

abc.azurecr.io/library/web
abd.azurecr.io/web

Reply:
Never mind, I see the bt is from the original name. 

Another corner case is users may write something like:
```
From registry.hub.docker.com/library/Ubuntu
From Ubuntu
```
But I think dup is OK for such case.

Reply:
yeah that's a good corner case, I will make a note of that. We need to rewrite these libs eventually so I can use them in the next builder and they're not perfect at normalizing things right now